enigmatry-entry-blueprint-app/src/app/shared/validators/custom-validators.ts

```diff
@@ -1,28 +1,29 @@
 import { UntypedFormControl } from '@angular/forms';
-import { firstValueFrom } from 'rxjs';
+import { Observable, firstValueFrom } from 'rxjs';
 import { ValidatorsService as CustomValidatorsService } from './validators.service';
 
 export { ValidatorsService as CustomValidatorsService } from './validators.service';
 
-const productCodeIsUniqueValidator = async(control: UntypedFormControl, service: CustomValidatorsService)
-    : Promise<{ productCodeIsUnique: boolean } | null> => {
+const validate = async<T>(control: UntypedFormControl,
```

Comment:
Can we follow top/down reading/writing convention.
 productCodeIsUniqueValidator  is calling validate method so validate should be below, not above.